pontoon/base/tests/test_views.py

```diff
@@ -124,6 +126,40 @@ def test_not_authed_public_project(self):
         # templates.
 
 
+class TranslateLocaleAgnosticTests(TestCase):
+    def setUp(self):
+        locale = LocaleFactory.create(code='custom-locale')
+        project = ProjectFactory.create(slug='valid-project', locales=[locale])
+        ResourceFactory.create(project=project)
+
+    def test_user_authed(self):
+        """
+        User is authenticated and Userprofile.custom_homepage defined,
+        redirect to Userprofile.custom_homepage
+        """
+        with patch('django.contrib.auth.models.User') as user_mock:
```

Comment:
@maiquynhtruong I feel you don't need to mock this object, you can just use django TestClient and login method.
https://docs.djangoproject.com/en/2.0/topics/testing/tools/#django.test.Client.login

You can create during the setup phase and just later just make login call.